### PR TITLE
Don't apply role="gridcell" to blank days or role="row" to blank week

### LINKED ISSUE #5436: Fix Calendar Accessibility with Blank Dates
Related to #4068 

### Expected Behavior

When a user interacts with a calendar with "showAdjacentDays" = false, the blank days should not be a part of the table and/or there needs to be proper context given to those with screen readers.

### Actual Behavior

With a calendar with "showAdjacentDays" = false, a user needs to go through the blank dates in the table. The screen reader reads "blank" with these dates, which is confusing and provides no helpful information for context.

### Video exhibiting the issue

https://user-images.githubusercontent.com/38971752/125327043-b1fc0e80-e310-11eb-9b96-c1b2166bc3db.mov

### Steps to Reproduce

1. Using VoiceOver for Mac or another screen reader, open the story for Show Adjacent Days Calendar in Storybook
2. Use keyboard commands to navigate to the days in the table for the "false" state and interact with the blank spots

### Your Environment

- Grommet version: v2.17.3
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): Macbook Pro Laptop

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Removes use of role="gridcell" on blank Calendar days and role="row" from weeks that are completely blank to improve experience of screen readers.

Removing "gridcell" on blank days ensures a screen reader user doesn't need to navigate over these days.

The removal of role="row" on blank weeks is needed for screen reader to properly calculate number of rows in a table. If not, a completely blank week will still be included in the row count but not accessible by screen reader which is confusing. For example "Row 1 of 6" where only 5 rows are able to be accessed because on week is empty.

#### Where should the reviewer start?
src/js/components/Calendar/Calendar.js

#### What testing has been done on this PR?
Tested with VoiceOver locally.

#### How should this be manually tested?
Test on showAdjacentDays story with VoiceOver and arrow keys. Should not be able to access blank days.

#### What are the relevant issues?
Closes #5436 

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Yes. Improved behavior for screen reader users by removing blank days from navigable Calendar days.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by amandadupell — APPROVED** [review]
looks good to me!!

File: src/js/components/Calendar/Calendar.js [hunk 1]
```diff
@@ -640,6 +640,8 @@ const Calendar = forwardRef(
     let day = new Date(displayBounds[0]);
     let days;
     let firstDayInMonth;
+    // if a week is filled with all blank days, we don't want to set role="row"
```
Comment (halocline) [inline src/js/components/Calendar/Calendar.js:643]: Feel like this comment may be more helpful where the logic is implemented on L793 and this comment could either be eliminated or read
```suggestion
    // a week containing only blank days
```
if you feel like it is helpful to be defined.

File: src/js/components/Calendar/Calendar.js [hunk 2]
```diff
@@ -776,11 +786,14 @@ const Calendar = forwardRef(
           );
         }
       }
-
       day = addDays(day, 1);
     }
     weeks.push(
-      <StyledWeek role="row" key={day.getTime()} fillContainer={fill}>
+      <StyledWeek
+        role={!blankWeek ? 'row' : undefined}
```
Comment (halocline) [inline src/js/components/Calendar/Calendar.js:793]:
```suggestion
        // if a week contains only blank days, for screen reader accessibility we don't want to set role="row"
        role={!blankWeek ? 'row' : undefined}
```

**Review by halocline — APPROVED** [review]
Functionality is very nice.

Minor suggestion regarding placement of code comments. Soft opinion.

File: src/js/components/Calendar/Calendar.js (hunk 2 above)
Comment (taysea) [reply src/js/components/Calendar/Calendar.js:793]: Included this in recent push.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**